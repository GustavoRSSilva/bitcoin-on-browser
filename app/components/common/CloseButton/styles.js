import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 100%;
  height: 20px;
  margin-top: 20px;
  padding: 0 20px;

  > button {
    float: right;
    width: 20px;
    height: auto;
    padding: 0;

    &:hover {
      cursor: pointer;
    }

    > img {
      width: 100%;
      height: 100%;
    }
  }
`;
