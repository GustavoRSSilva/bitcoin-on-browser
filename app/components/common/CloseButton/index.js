/**
 *
 * CloseButton
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import CloseButtonSVG from 'assets/img/close_button.svg';

import { Wrapper } from './styles';

function CloseButton(props) {
  const { onClick } = props;
  return (
    <Wrapper>
      <button onClick={onClick} onKeyDown={onClick}>
        <img src={CloseButtonSVG} alt="close_button" />
      </button>
    </Wrapper>
  );
}

CloseButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default CloseButton;
