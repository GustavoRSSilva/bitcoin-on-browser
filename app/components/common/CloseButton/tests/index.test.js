import React from 'react';
import { shallow } from 'enzyme';

import CloseButton from '../index';

describe('<CloseButton />', () => {
  it('renders 1 <CloseButton /> component', () => {
    const component = shallow(<CloseButton onClick={() => null} />);
    expect(component).toHaveLength(1);
  });

  it('correctly dispatch event on click', () => {
    const mockCallBack = jest.fn();

    const component = shallow(
      <CloseButton onClick={mockCallBack}>
        <span>submit</span>
      </CloseButton>,
    );
    component.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('correctly match the snapshot', () => {
    const component = shallow(<CloseButton onClick={() => null} />);
    expect(component).toMatchSnapshot();
  });
});
