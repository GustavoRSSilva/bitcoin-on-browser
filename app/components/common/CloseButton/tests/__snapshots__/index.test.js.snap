// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton /> correctly match the snapshot 1`] = `
<styles__Wrapper>
  <button
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <img
      alt="close_button"
      src="IMAGE_MOCK"
    />
  </button>
</styles__Wrapper>
`;
